streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -345,8 +343,17 @@ private SinkNodeFactory(final String name,
         }
     }
 
+    public void setTopologyName(final String namedTopology) {
+        Objects.requireNonNull(namedTopology, "named topology can't be null");
+        if (this.namedTopology != null) {
+            log.error("Tried to reset the namedTopology to {} but it was already set to {}", namedTopology, this.namedTopology);
+            throw new IllegalStateException("NamedTopology has already been set to " + this.namedTopology);
+        }
+        this.namedTopology = namedTopology;
+    }
+
     // public for testing only
-    public synchronized final InternalTopologyBuilder setApplicationId(final String applicationId) {
```

Comment:
Why remove synchronization here?

Reply:
I looked around and can't imagine why we would ever need it. `setApplicationId` should only ever be called once, from a single location/thread

Reply:
Ack.